ReportingServicesTools/Functions/CatalogItems/Rest/Get-RsRestCacheRefreshPlan.ps1

```diff
@@ -0,0 +1,117 @@
+        .EXAMPLE
+            Get-RsRestItem -RsReport "/MyReport" -WebSession $session
+            Description
+            -----------
+            Fetches item object "MyReport" catalog item found in "/" folder from the Report Server located at specificed WebSession object.
+        .EXAMPLE
+            Get-RsRestItem -RsReport "/MyReport" -ReportPortalUri http://myserver/reports
+            Description
+            -----------
+            Fetches catalog item "MyReport" catalog item found in "/" folder from the Report Server located at http://myserver/reports.
+        .EXAMPLE
+            Get-RsRestItem -RsReport "/Finance" -ReportPortalUri http://myserver/reports
+            Description
+            -----------
+            Fetches item "Finance" catalog item, which is a Folder object found in "/" folder from the Report Server located at http://myserver/reports.
+    #>
+
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $True, ValueFromPipelineByPropertyName = $true)]
+        [Alias('ItemPath','Path', 'RsItem')]
+        [string]
+        $RsReport,
+
+        [string]
+        $ReportPortalUri,
+
+        [Alias('ApiVersion')]
+        [ValidateSet("v2.0")]
+        [string]
+        $RestApiVersion = "v2.0",
+
+        [Alias('ReportServerCredentials')]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Microsoft.PowerShell.Commands.WebRequestSession]
+        $WebSession
+    )
+    Begin
+    {
+        $WebSession = New-RsRestSessionHelper -BoundParameters $PSBoundParameters
+        $ReportPortalUri = Get-RsPortalUriHelper -WebSession $WebSession
+        $catalogItemsUri = $ReportPortalUri + "api/$RestApiVersion/PowerBIReports({0})/CacheRefreshPlans"
+    }
+    Process
+    {
+        try
+        {
+            Write-Verbose "Fetching metadata for $RsReport..."
+            if ($Credential -ne $null)
+            {
+                $Report = Get-RsItem -ReportPortalUri $ReportPortalUri -RsItem $RsReport -WebSession $WebSession -Credential $Credential -Verbose:$false
+            }
+            else
+            {
+                $Report = Get-RsItem -ReportPortalUri $ReportPortalUri -RsItem $RsReport -WebSession $WebSession -Verbose:$false
+            }
+
+            Write-Verbose "Fetching CacheRefreshPlans for $RsReport..."
+            $catalogItemsUri = [String]::Format($catalogItemsUri, $Report.Id)
+            if ($Credential -ne $null)
+            {
+                $response = Invoke-RestMethod -Uri $catalogItemsUri -Method Get -WebSession $WebSession -Credential $Credential -Verbose:$false
+            }
+            else
+            {
+                $response = Invoke-RestMethod -Uri $catalogItemsUri -Method Get -WebSession $WebSession -UseDefaultCredentials -Verbose:$false
+            }
+
+            $item = $response.value
+            return [pscustomobject]@{
```

Comment:
does this automatically works if there is more than one cacherfreshplan for the report? how does iterate among all the possible cache refresh plans to populate a different $item.id?

Reply:
I have added a `foreach` loop to load all the plans returned into a `$CacheRefreshPlans` object, to be returned.